src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -364,6 +364,14 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
     });
   }
 
+  /**
+   * Handles canceling existing session event without saving changes.
+   */
+  cancelExistingSessionHandler(): void {
+    this.sessionEditFormModel.isEditable = false;
+    this.loadFeedbackSession();
```

Comment:
Do you really need to reload the session after cancelling any changes?

Reply:
Yes, otherwise the discarded changes aren't reflected until a page refresh.

Reply:
Is there not a way to roll back the changes in the front-end? This line here will re-fetch the course and feedback session data from the back-end, which should not be necessary because there is no new information to be fetched.

Reply:
Instead of loading it again, you can save the untouched session somewhere and restore it here.

Reply:
I added a "beforeEdit" object to store the session instructions on the front-end. Another entry could be added very easily to the "beforeEdit" object, if that entry was made editable and needed to be saved (for example "Session name"). 

I thought about saving every entry, just in case one was made editable in the future, but I thought I'd mention it to you guys first.

Reply:
Done! I made a "beforeEditModel" based on the SessionEditFormModel, so that it may be easier to maintain if other entries are made editable. Still not sure if this is the best way though, as I also considered having a "beforeEditIntructions" variable instead (to store just the unchanged instructions). Let me know what you think

Reply:
I think it is better to store the original `FeedbackSession` and use `getSessionEditFormModel` to generate a model for the form.

Reply:
Ok, I've used `FeedbackSession` and `getSessionEditFormModel ` to generate the original model on discard changes. I also disabled the lint no null assertion for one line, similar to how it was done in `discardExistingQuestionHandler()`